interactions/client/bot.py

```diff
@@ -1539,6 +1541,162 @@ async def _logout(self) -> None:
+                custom_ids.extend([c.custom_id for c in components.components])  # noqa
+            elif isinstance(components, str):
+                custom_ids.append(components)
+
+        if messages:
+            if isinstance(messages, Message):
+                messages_ids.append(int(messages.id))
+            elif isinstance(messages, list):
+                for message in messages:
+                    if isinstance(message, Message):
+                        messages_ids.append(int(message.id))
+                    else:
+                        messages_ids.append(int(message))
+            else:  # account for plain ints, string, or Snowflakes
+                messages_ids.append(int(messages))
+
+        def _check(ctx: ComponentContext) -> bool:
+            if custom_ids and ctx.data.custom_id not in custom_ids:
+                return False
+            if messages_ids and int(ctx.message.id) not in messages_ids:
+                return False
+            return check(ctx) if check else True
+
+        return await self.wait_for("on_component", check=_check, timeout=timeout)
+
+    async def wait_for_modal(
+        self,
+        modals: Union[Modal, str, List[Union[Modal, str]]],
+        check: Optional[Callable[[CommandContext], bool]] = None,
+        timeout: Optional[float] = None,
+    ):
+        """
+        Waits for a modal to be interacted with, and returns the resulting context.
+
+        :param Union[Modal, str, List[Modal, str]] modals: The modal(s) to wait for
+        :param Callable check: A function or coroutine to call, which should return a truthy value if the data should be returned
+        :param Optional[float] timeout: How long to wait for the event before raising an error
+        :return: The context of the modal
+        :rtype: CommandContext
+        """
+        ids: List[str] = []
+
+        if isinstance(modals, Modal):
+            ids = [str(modals.custom_id)]
+        elif isinstance(modals, str):
+            ids = [modals]
+        elif isinstance(modals, list):
+            for modal in modals:
+                if isinstance(modal, Modal):
+                    ids.append(str(modal.custom_id))
+                elif isinstance(modal, str):
+                    modals.append(modal)
+
+        if not all(isinstance(id, str) for id in ids):
+            raise TypeError("No modals were passed!")
+
+        def _check(ctx: CommandContext):
+            if ids and ctx.data.custom_id not in ids:
+                return False
+            return check(ctx) if check else True
+
+        ctx: CommandContext = await self.wait_for("on_modal", check=_check, timeout=timeout)
+
+        # Ed requested that it returns a result similar to the decorator
+        fields: List[str] = []
+        for actionrow in ctx.data.components:  # discord is weird with this
+            if actionrow.components:
+                data = actionrow.components[0].value
+                fields.append(data)
```

Comment:
```suggestion
            if components := actionrow.components:
                fields.append(components[0].value)
```

Reply:
It's only one line, I think leaving it as is keeps it a slight bit clearer